From 970083678782b9ef9568fbfacc0102583eb67fa7 Mon Sep 17 00:00:00 2001
From: Evisceration <alexander.martinz@nameless-rom.org>
Date: Tue, 6 Jan 2015 14:19:43 +0100
Subject: [PATCH] frameworks: hardcode package name to android

  * It can happen, that another context is accessing these methods.
    For example Settings calls the PowerProfile for battery stats, getPackageName
    then returns "com.android.settings" instead of "android" and the xml file
    can not be found
  * Also change the logging format

Change-Id: I19273e67b54eec6d124c3274f35a39e8540f42b4
Signed-off-by: Evisceration <alexander.martinz@nameless-rom.org>
---
 services/core/java/com/android/server/MountService.java | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/services/core/java/com/android/server/MountService.java b/services/core/java/com/android/server/MountService.java
index d2a7d89..3c98aa3 100644
--- a/services/core/java/com/android/server/MountService.java
+++ b/services/core/java/com/android/server/MountService.java
@@ -1385,15 +1385,13 @@ class MountService extends IMountService.Stub
          */
         String storageListOverride = SystemProperties.get("ro.storage_list.override");
         if (!storageListOverride.isEmpty()) {
-            int tmp_id = resources.getIdentifier(storageListOverride, "xml",
-                                                 mContext.getPackageName());
-            if(tmp_id > 0) {
-                Slog.i(TAG, "readStorageListLocked: using storage list "
-                       + storageListOverride);
-                id = tmp_id;
-            } else {
-                Slog.e(TAG, "readStorageListLocked: could not retrieve storage list "
-                       + storageListOverride + " using default instead");
+            int tmpId = resources.getIdentifier(storageListOverride, "xml", "android");
+            if(tmpId > 0) {
+                Slog.i(TAG, "readStorageListLocked: using storage list \"" + storageListOverride + "\"");
+                id = tmpId;
+             } else {
+                Slog.e(TAG, "readStorageListLocked: could not retrieve storage list \""
+                       + storageListOverride + "\", using default instead");
             }
         }
         XmlResourceParser parser = resources.getXml(id);
-- 
1.9.1

